core/lib/snapshots_applier/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+use std::{collections::HashMap, fmt};
+
+use anyhow::Error;
+use async_trait::async_trait;
+use zksync_core::sync_layer::MainNodeClient;
+use zksync_dal::{ConnectionPool, SqlxError, StorageProcessor};
+use zksync_object_store::{ObjectStore, ObjectStoreError};
+use zksync_types::{
+    snapshots::{
+        SnapshotFactoryDependencies, SnapshotHeader, SnapshotRecoveryStatus, SnapshotStorageLog,
+        SnapshotStorageLogsChunk, SnapshotStorageLogsStorageKey,
+    },
+    zkevm_test_harness::k256::pkcs8::der::EncodeValue,
+    MiniblockNumber, H256,
+};
+use zksync_utils::bytecode::hash_bytecode;
+use zksync_web3_decl::{
+    jsonrpsee::core::ClientError as RpcError,
+    namespaces::{EthNamespaceClient, ZksNamespaceClient},
+};
+
+use crate::SnapshotApplierError::*;
+
+mod tests;
+
+#[derive(thiserror::Error, Debug)]
+pub enum SnapshotApplierError {
```

Comment:
Yeah, I didn't want to spend too much effort deciding which potentially rare errors are retryable as the worst case scenario is that it would do a meaningless retry a few times